charmhelpers/contrib/charmsupport/nrpe.py

```diff
@@ -509,6 +510,37 @@ def add_haproxy_checks(nrpe, unit_name):
         check_cmd='check_haproxy_queue_depth.sh')
 
 
+def add_openvswitch_checks(nrpe, unit_name):
+    """
+    Add checks for openvswitch
+
+    :param NRPE nrpe: NRPE object to add check to
+    :param str unit_name: Unit name to use in check description
+    """
+    enable_sudo_for_openvswitch_checks()
+    nrpe.add_check(
+        shortname='openvswitch',
+        description='Check Open vSwitch {%s}' % unit_name,
+        check_cmd='check_openvswitch.py')
+
+
+def enable_sudo_for_openvswitch_checks():
+    sudoers_dir = "/etc/sudoers.d"
+    sudoers_mode = 0o100440
+    ovs_sudoers_file = "99-check_openvswitch"
+    ovs_sudoers_entry = "nagios ALL=(root) NOPASSWD: /usr/bin/ovs-vsctl --format=json list Interface"
```

Comment:
It'd be nice to include a 'Juju owned - do not edit' header, if possible?

Reply:
+1